Sources/MapboxMaps/Foundation/Tracing.swift

```diff
@@ -0,0 +1,98 @@
+import Foundation
+import os.signpost
+import MapboxCommon
+
+private let subsystem = "com.mapbox.maps"
+
+internal extension OSLog {
+    @available(iOS 12, *)
+    private static let _poi = OSLog(subsystem: subsystem, category: .pointsOfInterest)
+    private static let _platform = OSLog(subsystem: subsystem, category: "platform")
+    private static let tracingEnabled: Bool = SettingsServiceFactory.getInstance(storageType: .nonPersistent).isTracingEnabled
+
+    static let platform: OSLog = tracingEnabled ? _platform : .disabled
+    static let poi: OSLog = {
+        if #available(iOS 12, *), tracingEnabled {
+            return ._poi
+        } else {
+            return .disabled
+        }
+    }()
+
+    func beginInterval(name: StaticString, beginMessage: String? = nil) -> SignpostInterval {
+        let interval = SignpostInterval(log: self, name: name)
+        interval.begin(message: beginMessage)
+        return interval
+    }
+
+    func signpostEvent(name: StaticString, message: String? = nil) {
+        if #available(iOS 12, *) {
+            signpost(.event, log: self, name: name, message: message)
+        }
+    }
+}
+
+internal struct SignpostInterval {
+    private enum SignpostType {
+        case begin
+        case end
+        case event
+
+        @available(iOS 12, *)
+        var asOSType: OSSignpostType {
+            switch self {
+            case .begin: return .begin
+            case .end: return .end
+            case .event: return .event
+            }
+        }
+    }
+    private typealias Call = (SignpostType, String?) -> Void
+    private let call: Call?
+
+    init(log: OSLog, name: StaticString) {
+        if #available(iOS 12, *) {
+            let id = OSSignpostID(log: log)
+            call = {
+                signpost($0.asOSType, log: log, name: name, signpostID: id, message: $1)
+            }
+        } else {
+            call = nil
+        }
+    }
+
+    fileprivate func begin(message: String? = nil) {
+        call?(.begin, message)
+    }
+
+    func end(message: String? = nil) {
+        call?(.end, message)
+    }
+
+    func event(message: String? = nil) {
+        call?(.event, message)
+    }
+}
+
+@available(iOS 12, *)
+private func signpost(_ type: OSSignpostType, log: OSLog, name: StaticString, signpostID: OSSignpostID = .exclusive, message: String? = nil) {
+    if let message = message {
+        os_signpost(type, log: log, name: name, signpostID: signpostID, "%s", message)
+    } else {
+        os_signpost(type, log: log, name: name, signpostID: signpostID)
+    }
+}
+
+
+private extension SettingsServiceInterface {
+    var isTracingEnabled: Bool {
+        do {
+            // Enable tracing in iOS sdk when signposts for GLNative are enabled.
+            let tracing = try get(key: "com.mapbox.tracing", type: Dictionary<String, String>.self).get()
+            return tracing[subsystem] == "platform"
```

Comment:
Is it the only supported value in here? Can it be something other than just a `platform`?

Reply:
As [`NSArgumentDomain`](https://developer.apple.com/documentation/foundation/nsargumentdomain) doesn't support arrays and dictionaries, may be we can reconsider that key structure?